test/server/cards/02_SHD/units/TarffulKashyyykChieftain.spec.ts

```diff
@@ -51,6 +52,31 @@ describe('Tarfful, Kashyyyk Chieftain', function() {
 
                 expect(context.isbAgent).toBeInZone('groundArena', context.player1);
                 expect(context.player1).toBeActivePlayer();
+
+                // Scenario 5: a friendly wookiee unit is attacked and survives
+                context.player1.moveCard(context.liberatedSlaves, 'groundArena');
+                context.player1.passAction();
+
+                context.player2.moveCard(context.atst, 'groundArena');
+                context.player2.clickCard(context.volunteerSoldier);
+                context.player2.clickCard(context.liberatedSlaves);
+
+                expect(context.player1).toBeAbleToSelectExactly([context.wampa, context.wroshyrTreeTender, context.atst]);
+                context.player1.clickCard(context.atst);
+
+                expect(context.atst.damage).toBe(3);
+
+                // Scenario 6: non-combat damage does not trigger Tarfful's ability
+                context.player1.passAction();
+
+                context.player2.clickCard(context.qira);
+                context.player2.clickPrompt('Deploy Qi\'ra');
+
+                expect(context.player1).toBeActivePlayer();
+                expect(context.player1).toHavePrompt('Choose an action');
+
+                // TODO: Add test with SHD Maul unit redirecting damage to a friendly wookiee underworld unit
+                // TODO: Add test with TWI Maul unit dealing damage to multiple friendly wooikee at the same time
```

Comment:
Actually, thinking about it let's add a third as well for the case I raised where Tarfful and another wookiee is attacked by Maul2 and only Tarfful is defeated